lib/internal/http2/core.js

```diff
@@ -2774,7 +2774,18 @@ function connect(authority, options, listener) {
   const protocol = authority.protocol || options.protocol || 'https:';
   const port = '' + (authority.port !== '' ?
     authority.port : (authority.protocol === 'http:' ? 80 : 443));
-  const host = authority.hostname || authority.host || 'localhost';
+  let host = '';
+
+  if (authority.hostname) {
+    host = authority.hostname;
+
+    if (host.startsWith('['))
```

Comment:
Maybe `host[0] === '['` ?
Ref: https://github.com/typescript-eslint/typescript-eslint/issues/285#issuecomment-463802241

Reply:
It is consistent with https://github.com/nodejs/node/blob/1ff9a49b7811f77919075132559adf105c3291b5/lib/internal/url.js#L1260 but I'm fine changing it if needed.

Reply:
Btw there seems to be a regression in terms of performance in the latest V8.

```
luigi@imac:bench$ node -v
v12.4.0
luigi@imac:bench$ node index.js 
[] x 206,332,202 ops/sec ±0.11% (99 runs sampled)
charAt() x 206,542,018 ops/sec ±0.10% (95 runs sampled)
startsWith() x 58,000,980 ops/sec ±0.33% (97 runs sampled)
Fastest is charAt(),[]
luigi@imac:bench$ ~/code/node/node -v       
v13.0.0-pre
luigi@imac:bench$ ~/code/node/node index.js 
[] x 109,972,754 ops/sec ±3.83% (81 runs sampled)
charAt() x 89,506,798 ops/sec ±2.71% (84 runs sampled)
startsWith() x 40,955,265 ops/sec ±1.35% (88 runs sampled)
Fastest is []
```